tiledb/sm/array_schema/array_schema.h

```diff
@@ -362,10 +364,10 @@ class ArraySchema {
   ArrayType array_type_;
 
   /** It maps each attribute name to the corresponding attribute object. */
-  std::unordered_map<std::string, Attribute*> attribute_map_;
+  std::unordered_map<std::string, const Attribute*> attribute_map_;
 
   /** The array attributes. */
```

Comment:
There's documentation about life spans that's needed here, since the `shared_ptr` is ensuring life for elements in the attribute map. A corresponding comment about the attribute map is also warranted.